### ISSUE TITLE
Random crashes when running aaa in combination with QtWebEngine

### ISSUE THREAD

**Issue by thestr4ng3r** [radareorg/radare2#11693]
Reproducer: https://github.com/thestr4ng3r/r2-webengine-crash (it's important to use a mips binary)
Run this a few times and it will crash at random places.
This was the cause of https://github.com/radareorg/cutter/issues/581

Interestingly, both asan and valgrind say absolutely nothing useful and I couldn't even produce a single crash when running with valgrind. This might be because valgrind makes everything run much slower if it is a timing-based issue.

**Comment by thestr4ng3r** [comment]
It might be realted to the anal hints that are set by `aav`. When that is disabled it doesn't crash.
Or if this line (which eventually uses the hints set by `aav`) is commented out, it doesn't crash as well:
https://github.com/radare/radare2/blob/883abe417f48e4f1803ecd2238418d45f061b3dd/libr/anal/op.c#L152
Take this info with a grain of salt since we are dealing with random crashes here and it might just not crash because the timing changes or whatever.

**Comment by Maijin** [comment]
It would explain why this is mostly seen in non-x86 binary though, because aav is triggered in aaa only when it's != x86

**Comment by thestr4ng3r** [comment]
exactly

**Comment by radare** [comment]
It is fixed in cutter by removing the qtwebengine?

**Comment by radare** [comment]
Any reproducer in r2?

**Comment by xarkes** [comment]
The only reproducer is here as said in the first comment: https://github.com/thestr4ng3r/r2-webengine-crash
So it's not *in* r2 but it happens because of r2 (or because of the QtWebEngine).
And yes, this is fixed in Cutter by removing the qtwebengine.

**Comment by XVilka** [comment]
Hopefully will get fixed soon and WebEngine will return back

**Comment by trufae** [comment]
asan builds dont use the thread sanitizer by default, and that's probably what makes `aaa` to crash, anyway, r2 is now way more thread-friendly than it was at the time of writing this issue and tasks are known to be buggy in some situations, so probably using the threadchannel while spawning a webserver on a clean thread is enough to solve all the problems related to this.

I've fixed racy bugs in `aav` already as well as other task-related bugs.

Closing as it's a known issue with a known solution and `iaito` doesnt uses or plans to use the bloated webkit or even tasks in there.
